ValveResourceFormat/IO/GltfModelExporter.cs

```diff
@@ -1055,6 +1023,50 @@ void WriteTexture(MaterialExtract.Channel channel, string gltfName, int index, i
             return material;
         }
 
+        private Image LinkImageMostEfficient(MaterialExtract.Channel channel, SkiaSharp.SKBitmap bitmap, ModelRoot model, string fileName)
+        {
+            var existingImage = model.LogicalImages.SingleOrDefault(i => i.Name == fileName);
+            if (existingImage is not null)
+            {
+                return existingImage;
+            }
+
+            Image image;
+            ProgressReporter?.Report($"Exporting texture: {fileName}");
+
+            if (!SatelliteImages)
+            {
+                image = model.CreateImage(fileName);
+                image.Content = TextureExtract.ToPngImageChannels(bitmap, channel);
+                return image;
+            }
+
+            var exportedTexturePath = Path.ChangeExtension(Path.Join(DstDir, fileName), "png");
+
+            // Write a 44 byte PNG file that can be replaced with the full image after linking to modelRoot.
+            // SharpGLTF loads images into memory, which isn't necessary with satellite images.
+            using (var fs = File.Open(exportedTexturePath, FileMode.Create))
```

Comment:
You can keep the fs open, rewind it and then write the full image.

Reply:
The CreateImage call just below this needs the file to be free.

Reply:
What if you specify FileShare.Read?

Reply:
That's good idea but can't get it to work.. I tried
`File.Open(exportedTexturePath, FileMode.Create, FileAccess.Write, FileShare.Read)`
`File.Open(exportedTexturePath, FileMode.Create, FileAccess.Write, FileShare.Write)`

Still throwing IOException.